File: app/src/main/java/app/endershrooms/thatcatapp/model/builders/BaseNewRequest.java
package app.endershrooms.thatcatapp.model.builders;

import com.squareup.moshi.Json;

public abstract class BaseNewRequest {

  @Json(name = "image_id")
  String imageId;

  @Json(name = "sub_id")
  String subId;

  BaseNewRequest(String imageId, String subId) {
    this.imageId = imageId;
    this.subId = subId;
  }

  BaseNewRequest(String imageId) {
    this.imageId = imageId;
  }

  public String getImageId() {
    return imageId;
  }

  public void setImageId(String imageId) {
    this.imageId = imageId;
  }

  public String getSubId() {
    return subId;
  }

  public void setSubId(String subId) {
    this.subId = subId;
  }
}

File: app/src/main/java/app/endershrooms/thatcatapp/model/builders/BaseNewRequest.kt
package app.endershrooms.thatcatapp.model.builders

import com.squareup.moshi.Json

abstract class BaseNewRequest(
        @field:Json(name = "image_id")
        var imageId: String,

        @field:Json(name = "sub_id")
        var subId: String?
) {

    constructor(imageId: String): this(imageId, null)
}
